ledger/src/blockstore_processor.rs

```diff
@@ -845,7 +849,9 @@ pub fn process_blockstore_from_root(
 ) -> result::Result<(), BlockstoreProcessorError> {
     let (start_slot, start_slot_hash) = {
         // Starting slot must be a root, and thus has no parents
-        assert_eq!(bank_forks.read().unwrap().banks().len(), 1);
+        if !opts.skip_bankforks_checks_for_wen_restart {
+            assert_eq!(bank_forks.read().unwrap().banks().len(), 1);
+        }
```

Comment:
This doesn't seem like the right entrypoint for the wen restart logic, hence the need to hack this flag. 

Among other things, `process_blockstore_from_root()` will play every possible fork in blockstore, we just want to play the slots in a given list (the ones we repaired). 

I think we should have a separate function that just calls `process_single_slot()` on every one of the repaired slots, should be much cleaner and simpler

Reply:
Hmm, re-reading Steve's comments, actually this will cause a problem, it may insert already present blocks. I'll change it, hold on.

Reply:
Re-checked, actually using process_single_slot is a good idea, I've changed it. Ready for another look.